src/logics/ArithLogic.cc

```diff
@@ -674,12 +691,7 @@ PTRef ArithLogic::mkTimes(vec<PTRef> && args) {
     PTRef tr = mkFun(s_new, std::move(args));
     // Either a real term or, if we constructed a multiplication of a
     // constant and a sum, a real sum.
-    if (isNumTerm(tr) || isPlus(tr) || isUF(tr) || isIte(tr))
-        return tr;
-    else {
-        auto termStr = pp(tr);
-        throw LANonLinearException(termStr.c_str());
-    }
+    return tr;
```

Comment:
The comment above is outdated at this point.

Reply:
Fixed